src/apis/impl/DataOwnerApiImpl.ts

```diff
@@ -71,24 +73,39 @@ export class DataOwnerApiImpl implements DataOwnerApi {
     keyPair?: { publicKey: string; privateKey: string }
   ): Promise<{ publicKey: string; privateKey: string }> {
     const dataOwnerId = this.getDataOwnerIdOf(user)
-
-    const { publicKey, privateKey } = keyPair ?? (await this._generateKeyPair())
-    const jwk = {
-      publicKey: spkiToJwk(hex2ua(publicKey)),
-      privateKey: pkcs8ToJwk(hex2ua(privateKey)),
-    }
-    await this.cryptoApi.cacheKeyPair(jwk)
-    await this.keyStorage.storeKeyPair(`${dataOwnerId}.${publicKey.slice(-32)}`, jwk)
-
     const dataOwner = await this.cryptoApi.getDataOwner(dataOwnerId).catch((e) => {
       throw this.errorHandler.createErrorFromAny(e)
     })
+
     if (dataOwner == null) {
       throw this.errorHandler.createErrorWithMessage(
         `The current user is not a data owner. You must be either a patient, a device or a healthcare professional to call this method.`
       )
     }
 
+    let jwks: { privateKey: JsonWebKey; publicKey: JsonWebKey } | undefined
+    if (keyPair === undefined && !overwriteExistingKeys && dataOwner.dataOwner.publicKey !== undefined) {
```

Comment:
To simplify this, I wonder if you don't have a "getPublicKeys" for a dataOwner in the icc-api but I'm not sure. If not, that could be interesting to have such a method.